plugin/documents.py

```diff
@@ -176,6 +179,7 @@ def _setup(self) -> None:
         self._stored_selection = []
         self._sighelp = None  # type: Optional[SigHelp]
         self._lightbulb_line = None  # type: Optional[int]
+        self._diagnostics_by_config = []  # type: List[Tuple[SessionBufferProtocol, List[Diagnostic]]]
```

Comment:
Maybe call it `diagnostics_for_selection` since in this context it looks like diagnostics in the whole file.

But then I suppose `_actions_by_config` could also be renamed.